cpp/tests/streams/interop_test.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/default_stream.hpp>
+
+#include <cudf/interop.hpp>
+#include <cudf/table/table_view.hpp>
+
+#include <dlpack/dlpack.h>
+
+struct dlpack_deleter {
+  void operator()(DLManagedTensor* tensor) { tensor->deleter(tensor); }
+};
+
+struct DLPackTest : public cudf::test::BaseFixture {};
+
+TEST_F(DLPackTest, ToDLPack)
```

Comment:
Should these two tests contain assertions, such as `EXPECT_NO_THROW` ?
Irrelevant to this PR, I noticed that many other tests out there under `cpp/tests/streams` don't have assertions. I'm curious if there's any special consideration for that.

Reply:
Stream tests only verify if the stream is being forwarded correctly. We do not check functional correctness here; the test tracks the test stream passed and verifies that it is used for all device operations.

Reply:
Thanks. I guess my question is what would be a signal of problem for these tests. Say I artificially introduced a bug where one device operation is on a stream separate from others. Do we expect `cudf::to_dlpack()` to error out?

Reply:
Ah sorry, I misunderstood what you were asking. Yes, if we use a stream other than `cudf::test::get_default_stream()` for a device operation anywhere in the test, a runtime error is thrown. CUDA APIs are overloaded to check the passed stream and either raise an exception for an unexpected stream, or forward arguments to the original function.
Reference: https://github.com/rapidsai/cudf/blob/f54c1a5ad34133605d3b5b447d9717ce7eb6dba0/cpp/tests/utilities/identify_stream_usage.cpp#L135
TLDR: If we invoke `cudf::to_dlpack(empty, cudf::get_default_stream())` in this test, a runtime error is thrown.